src/EventCalendar.js

```diff
@@ -65,120 +104,526 @@ class EventCalendar {
+    }
+
+    /**
+     * Make screen readers announce the `message` by updating a hidden aria-live region within the template
+     * 
+     * @param {string} message The text to be announced by the screen reader
+     */
+    updateLiveRegion (message) {
+        let _self = this;
+
+        _self.$ariaLiveRegion.text(message);
+    }
+
+    /**
+     * Style a date which should be removed from the list of active dates
+     * 
+     * @param {moment} target The date to style
+     */
+    styleToDel (target) {
+        let _self = this,
+            $elem = _self.clndr.element.find('[data-day="' + target.format(_self.dateFormat) + '"]'),
+            ariaLabel = 'Removed. Event will no longer repeat on ' + target.format(_self.dateFormatLong);
+
+        $elem.attr('aria-label', ariaLabel)
+             .parent()
+             .removeClass('event-add')
+             .addClass('event-del');
+
+        // Announce the new status through a live region (instead of the previous status)
+        _self.updateLiveRegion(ariaLabel);
+    }
+
+    /**
+     * Style a date which will be affected by the recurrence pattern and added to the list of active dates
+     * 
+     * @param {moment} target The date to style
+     */
+    styleRepeatOn (target) {
+        let _self = this,
+            $elem = _self.clndr.element.find('[data-day="' + target.format(_self.dateFormat) + '"]');
+
+        $elem.attr('aria-label', 'Event will repeat on ' + target.format(_self.dateFormatLong) + ' based on the chosen repeat pattern')
+             .parent()
+             .addClass('event-repeat');
+    }
+
+    /**
+     * Remove styling from a date and reset it to the 'neutral' state
+     * 
+     * @param {moment} target The date to style
+     */
+    styleClear (target) {
+        let _self = this,
+            $elem = _self.clndr.element.find('[data-day="' + target.format(_self.dateFormat) + '"]');
+
+        $elem.attr('aria-label', target.format(_self.dateFormatLong) + '.')
+             .parent()
+             .removeClass('event-add event-del event-repeat');
+
+        // Announce the new status through a live region (instead of the previous status)
+        _self.updateLiveRegion('Unselected. Event will not repeat on ' + target.format(_self.dateFormatLong));
+    }
+
+    /**
+     * Restore all dates to their initial state (when the calendar was initialised)
+     */
+    styleClearAll (target) {
+        let _self = this,
+            $elems = _self.clndr.element.find('.day-contents');
+
+            $elems.each(function() {
+                let $elem = $(this),
+                    $elemParent = $elem.parent(),
+                    date = moment($elem.data('day')).format(_self.dateFormatLong),
+                    ariaLabel = 'Unselected';
+
+                // If event was passed in the `events` array, it will be reset back to `.selected` and needs labelling
+                if ($elemParent.attr('class').indexOf('event') >= 0) {
```

Comment:
I normally use a `!== -1` when using `indexOf` which is why this caught my attention, and because I am sad I made this https://jsperf.com/pulsar-indexof/1, it turns out `!== -1` is 7% faster than `>= 0`, pre-optimisation is the root of all evil, but I thought I'd share.